packages/devtools_app/lib/src/screens/memory/memory_controller.dart

```diff
@@ -239,6 +239,19 @@ class MemoryController extends DisposableController
   MemoryController() {
     memoryTimeline = MemoryTimeline(offline);
     memoryLog = MemoryLog(this);
+
+    // Update the chart when the memorySource changes.
+    addAutoDisposeListener(memorySourceNotifier, () async {
```

Comment:
did this code accidentally slip into this PR? it seems unrelated from the rest

Reply:
Now, when notifications can be accesses from controllers, this code can be moved from UI(chart) to the controller.
Actually this is why I started this PR, and it resulted in refactoring for notifications.

I can separate it, if it is needed for something.